lib/money/money.rb

```diff
@@ -83,16 +84,16 @@ def resolve_currency
   end
   default_settings
 
-  def initialize(value = 0, currency = nil)
-    raise ArgumentError if value.respond_to?(:nan?) && value.nan?
+  def initialize(value, currency)
```

Comment:
Are you sure its ok to change the signature of this method? I know it should only be called by the `#new` on the class, but, in the off chance that this is invoked by anyone doing something crazy, this would be a breaking change, right? Should we do a version bump maybe?

Reply:
We're not released on Rubygems and pre 1.0, the only breakage I'd really worry about is Shopify core.